app/logical/post_query_builder.rb

```diff
@@ -502,10 +507,19 @@ def should(*args)
       order.push({_score: :desc})
 
     when "random"
-      must.push({function_score: {
-          query: {match_all: {}},
-          random_score: {},
-      }})
+      if q[:random].present?
```

Comment:
I'd recommend
```rb
        must.push({function_score: {
            query: {match_all: {}},
            random_score: {},
            boost_mode: :replace
        }})
```
for both cases. (Note that `NilClass#to_i` is defined and always returns zero.)

Reply:
I need to revise this anyways, because if you specify a seed it uses a different sorting field than if you don't specify one. So splitting them does make sense. For the future I need to specify a field as well and set it to the `_seq_no` field or the `id` field so that it doesn't load larder less efficient fields into memory.